machine/v1beta1/types_vsphereprovider.go

```diff
@@ -172,6 +179,15 @@ type NetworkDeviceSpec struct {
 	AddressesFromPools []AddressesFromPool `json:"addressesFromPools,omitempty"`
 }
 
+// VSphereDisk describes additional disks for vSphere.
+type VSphereDisk struct {
+	// The device name exposed. Used to identify the disk.
```

Comment:
if not defined, what is the behavior? is a disk created for you?

Reply:
If you do not specify a device name, it will still create the disk with the size you defined.  The deviceName  field is to be used by admin to sort of remind them what disk is meant to be used for (if they so desire).